linera-core/src/unit_tests/worker_tests.rs

```diff
@@ -386,54 +388,55 @@ where
 
 #[test(tokio::test)]
 async fn test_memory_handle_block_proposal_ticks() {
-    let client = MemoryStoreClient::make_test_client(None).await;
+    let client = MemoryStoreClient::make_test_client(None, TestClock::new()).await;
     run_test_handle_block_proposal_ticks(client).await;
 }
 
 #[cfg(feature = "rocksdb")]
 #[test(tokio::test)]
 async fn test_rocks_db_handle_block_proposal_ticks() {
     let _lock = ROCKS_DB_SEMAPHORE.acquire().await;
-    let client = RocksDbStoreClient::make_test_client(None).await;
+    let client = RocksDbStoreClient::make_test_client(None, TestClock::new()).await;
     run_test_handle_block_proposal_ticks(client).await;
 }
 
 #[cfg(feature = "aws")]
 #[test(tokio::test)]
 async fn test_dynamo_db_handle_block_proposal_ticks() {
-    let client = DynamoDbStoreClient::make_test_client(None).await;
+    let client = DynamoDbStoreClient::make_test_client(None, TestClock::new()).await;
     run_test_handle_block_proposal_ticks(client).await;
 }
 
 #[cfg(feature = "scylladb")]
 #[test(tokio::test)]
 async fn test_scylla_db_handle_block_proposal_ticks() {
-    let client = ScyllaDbStoreClient::make_test_client(None).await;
+    let client = ScyllaDbStoreClient::make_test_client(None, TestClock::new()).await;
     run_test_handle_block_proposal_ticks(client).await;
 }
 
-async fn run_test_handle_block_proposal_ticks<S>(client: S)
+async fn run_test_handle_block_proposal_ticks<C>(client: DbStoreClient<C, TestClock>)
 where
-    S: Store + Clone + Send + Sync + 'static,
-    ViewError: From<S::ContextError>,
+    C: KeyValueStoreClient + Clone + Send + Sync + 'static,
+    ViewError: From<<C as KeyValueStoreClient>::Error>,
+    <C as KeyValueStoreClient>::Error:
+        From<bcs::Error> + From<DatabaseConsistencyError> + Send + Sync + serde::ser::StdError,
 {
     let key_pair = KeyPair::generate();
     let balance: Amount = Amount::from_tokens(5);
     let balances = vec![(ChainDescription::Root(1), key_pair.public(), balance)];
     let epoch = Epoch::ZERO;
+    let clock = client.clock.clone();
     let (committee, mut worker) = init_worker_with_chains(client, balances).await;
 
     {
         let block_proposal = make_first_block(ChainId::root(1))
-            .with_timestamp(
-                Timestamp::now().saturating_add_micros(TEST_GRACE_PERIOD_MICROS + 1_000_000),
-            )
+            .with_timestamp(Timestamp::from(TEST_GRACE_PERIOD_MICROS + 1_000_000))
             .into_simple_proposal(&key_pair);
         // Timestamp too far in the future
         assert!(worker.handle_block_proposal(block_proposal).await.is_err());
     }
 
-    let block_0_time = Timestamp::now().saturating_add_micros(TEST_GRACE_PERIOD_MICROS);
+    let block_0_time = Timestamp::from(TEST_GRACE_PERIOD_MICROS);
```

Comment:
nit: maybe `from_now` or `after`?

Reply:
No, this is really `TEST_GRACE_PERIOD_MICROS` after the Unix epoch, because all the tests now start at the Unix epoch by default.

Reply:
gotcha